src/governance.ts

```diff
@@ -72,6 +75,15 @@ export function vetoCouncilElectionOrSlashHandler(
   entity.transactionHash = event.transaction.hash;
   entity.proposalType = "VetoCouncilElectionOrSlash";
   entity.save();
+
+  createActivity(
+    event,
+    "Veto",
+    event.params.proposer.toHexString(),
+    event.params.proposalId.toString(),  // proposalId
+    null, // votes
+    null, // glowAmount
+  );
```

Comment:
One think I’m not 100 % sure of is if I’m creating activities with the correct type in this file @0xSimbo

Reply:
I think that the wording on the activity is a bit confusing. You might want to change the name of the activity to VetoProposalCreation because VetoCouncil members can veto proposals which is a completely different action. You can also get the number of nominations used on the proposal from the event I believe.

```/**
     * @notice Emitted when a Veto Council Election or Slash proposal is created
     * @param proposalId the id of the proposal
     * @param proposer the address of the proposer
     * @param oldAgent the address of the old agent
     * @param newAgent the address of the new agent
     * @param slashOldAgent whether or not to slash the old agent
     * @param nominationsUsed the amount of nominations used
     */
    event VetoCouncilElectionOrSlash(
        uint256 indexed proposalId,
        address indexed proposer,
        address oldAgent,
        address newAgent,
        bool slashOldAgent,
        uint256 nominationsUsed
    );```
    
    Votes are different than nominations used on propsals. Votes come from ratify or reject activities while nomination spent comes from creating proposals and using nominations on proposals.

Reply:
You can also find the rest of the events here:
https://github.com/glowlabs-org/glow-contracts/blob/dev/src/interfaces/IGovernance.sol

Reply:
Sounds good! do you think the type of this action could just be "Create" since the user is creating a proposal? (to be consistent with the other action types)

And on the frontend, we can get the proposal type from proposal id:
 proposalType: 'VetoCouncilElectionOrSlash',
and in the UI say something like "0x123abc created a proposal in veto council election or slash using x nominations"

otherwise i can make all the actions a bit more descriptive. e.g.: instead of "Create" use "GrantProposalCreation" etc.